src/webview/VSCodeKaotoEditorChannelApi.ts

```diff
@@ -1,9 +1,35 @@
 import { KaotoEditorChannelApi } from '@kaoto/kaoto';
 import { ISettingsModel, NodeLabelType, SettingsModel } from '@kaoto/kaoto/models';
+import { BackendProxy } from '@kie-tools-core/backend/dist/api';
+import { NotificationsChannelApi } from "@kie-tools-core/notifications/dist/api";
+import { ResourceContentService, WorkspaceChannelApi } from "@kie-tools-core/workspace/dist/api";
+import { I18n } from '@kie-tools-core/i18n/dist/core';
 import { DefaultVsCodeKieEditorChannelApiImpl } from '@kie-tools-core/vscode-extension/dist/DefaultVsCodeKieEditorChannelApiImpl';
+import { VsCodeI18n } from '@kie-tools-core/vscode-extension/dist/i18n';
+import { VsCodeKieEditorController } from '@kie-tools-core/vscode-extension/dist/VsCodeKieEditorController';
+import { JavaCodeCompletionApi } from '@kie-tools-core/vscode-java-code-completion/dist/api';
 import * as vscode from 'vscode';
+import { VsCodeKieEditorCustomDocument } from '@kie-tools-core/vscode-extension/dist/VsCodeKieEditorCustomDocument';
+import * as path from 'path';
 
 export class VSCodeKaotoEditorChannelApi extends DefaultVsCodeKieEditorChannelApiImpl implements KaotoEditorChannelApi {
+
+  private readonly currentEditedDocument: vscode.TextDocument | VsCodeKieEditorCustomDocument;
+
+  constructor(
+    editor: VsCodeKieEditorController,
+    resourceContentService: ResourceContentService,
+    workspaceApi: WorkspaceChannelApi,
+    backendProxy: BackendProxy,
+    notificationsApi: NotificationsChannelApi,
+    javaCodeCompletionApi: JavaCodeCompletionApi,
+    viewType: string,
+    i18n: I18n<VsCodeI18n>
+  ) {
+    super(editor, resourceContentService, workspaceApi, backendProxy, notificationsApi, javaCodeCompletionApi, viewType, i18n);
+    this.currentEditedDocument = editor.document.document;
```

Comment:
[context] I created an issue for this https://github.com/apache/incubator-kie-tools/issues/2584